.github/workflows/wasm.yaml

```diff
@@ -126,13 +126,50 @@ jobs:
             name: Inspector Build
             path: tools/inspector/dist
 
+  ghPagesPreview:
+    runs-on: ubuntu-20.04
+    needs: [wasmQt6, inspector]
+    name: Publish Wasm on Github Pages
+    env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} #github.ref_name
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with: 
+          path: _site
+          ref: 'refs/heads/gh-pages'
+
+      - name: Download a Build Artifact
+        uses: actions/download-artifact@v2.0.8
+        with:
+          name: WebAssembly Build Qt6
+          # Destination path
+          path: _site/branches/${{github.ref_name}}
+      - name: Download a Build Artifact
+        uses: actions/download-artifact@v2.0.8
+        with:
+          name: Inspector Build
+          path: _site/branches/${{github.ref_name}}/inspector
+      - name: Uploading site Build
```

Comment:
we need to rewrite the git history otherwise we will create billions commits

Reply:
This is currently only generating a .zip artifact to check if the structure is correct.
But the idea is to checkout the gh-pages branch and then just amend the commit. 
Removing the "Cecilapp/GitHub-Pages-deploy@3.1.0" which will just add an extra commit :)